libs/common/types.hpp

```diff
@@ -112,31 +123,42 @@ namespace iroha {
     return std::string(source.begin(), source.end());
   }
 
-  // Deserialize hex string to array
-  template <size_t size>
-  inline void hexstringToArray(const std::string& string, blob_t<size>& array) {
-    auto bytes = hex2bytes(string);
-    std::copy(bytes.begin(), bytes.end(), array.begin());
-  }
-
   /**
    * Convert string of raw bytes to printable hex string
    * @param str
    * @return
    */
   inline std::string bytestringToHexstring(std::string str) {
-    std::string res(str.size() * 2, 0);
-    uint8_t front, back;
-    auto ptr = str.data();
-    for (uint32_t i = 0, k = 0; i < str.size(); i++) {
-      front = (uint8_t) (ptr[i] & 0xF0) >> 4;
-      back = (uint8_t) (ptr[i] & 0xF);
-      res[k++] = code[front];
-      res[k++] = code[back];
+    std::stringstream ss;
+    ss << std::hex << std::setfill('0');
+    for(const auto &c : str) {
+      ss << std::setw(2) << static_cast<int>(c);
     }
-    return res;
+    return ss.str();
   }
 
+  /**
+   * Convert printable hex string to string of raw bytes
+   * @param str
+   * @return
+   */
+  inline nonstd::optional<std::string> hexstringToBytestring(std::string str) {
+    if (str.empty() or str.size() % 2 != 0) {
+      return nonstd::nullopt;
+    }
+    std::string result(str.size() / 2, 0);
+    for (size_t i = 0; i < result.length(); ++i) {
+      std::string byte = str.substr(i * 2, 2);
```

Comment:
Can be either replaced with check string is hex and grub via istringstream (imo better & faster)
or at least don't (de-)allocate on every byte std::string (e.g char[3] can help)